src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Formatting/Passes/RazorFormattingPass.cs

```diff
@@ -403,6 +403,13 @@ private bool FormatBlock(FormattingContext context, RazorSourceDocument source,
             {
                 var openBraceLineNumber = openBraceNode.GetLinePositionSpan(source).Start.Line;
                 var openBraceLine = source.Text.Lines[openBraceLineNumber];
+                // The open brace node might actually start with a newline on the line before, which could be blank,
```

Comment:
>// The open brace node might actually start with a newline on the line before [](http://example.com/codeflow?start=16&length=77)

Does this not happen sometimes too? Just curious if it's consistently that way or inconsistent.

Reply:
Yeah, there are existing tests that don't go into the `while`, and there is a feature flag for the compiler called "FUSE" which switches its generation entirely, and results in very different whitespace handling in the tree.